pkg/config/config.go

```diff
@@ -130,21 +132,48 @@ func (c *ConfigInfo) writeToFile() error {
 }
 
 // SetConfiguration modifies Odo configurations in the config file
-// as of now only being used for updatenotification
-func (c *ConfigInfo) SetConfiguration(parameter string, value bool) error {
+// as of now being used for timeout, updatenotification
+func (c *ConfigInfo) SetConfiguration(parameter string, value interface{}) error {
 	switch parameter {
+
+	case "timeout":
+		typedval, ok := value.(int)
+		if typedval == 0 {
+			fmt.Println("Cannot set timeout to 0, minimum value is 1")
+			return errors.Errorf("type assertion error: unable to set")
+		}
+		if ok != true {
+			return errors.Errorf("type assertion error: unable to set %s", parameter)
+		}
+		c.OdoSettings.Timeout = typedval
```

Comment:
Anyone that wants config would do config.New which reads a fresh from config file on every call right?
why is this required?

Reply:
towards the end of this function it writes into the config file and the next time it reads from there

Reply:
also cfg.New() is reading from config file if file exists

Reply:
https://github.com/redhat-developer/odo/blob/master/pkg/config/config.go#L75

Reply:
> Anyone that wants config would do config.New which reads a fresh from config file on every call right?
nope, if config file exist then it reads the existing config file 
https://github.com/redhat-developer/odo/blob/master/pkg/config/config.go#L75
> why is this required?
at the end of this function it gets written into the config file

Reply:
> towards the end of this function it writes into the config file and the next time it reads from there

Ok now makes sense